File: Assignment1/Problem5/C/main.cpp
#include <iostream>

int binarySearch(int arr[], int value, int left, int right);

int main() {
    clock_t start, end;
    //int size = 128;
    int size = 512;
    //int size = 2048;
    //int size = 8192;
    //int size = 32768;
    //int size = 131072;
    //int size = 524288;
    //int size = 2097152;
    int *array = new int[size];
    for (int i = 0; i < size; i++)
    {
        array[i]=0;
    }
    start = clock();
    for (int i = 0; i<10; i++)
    {
        binarySearch(array, 1, 0, size-1);
    }
    end = clock();
    double time = double(end-start)/CLOCKS_PER_SEC;
    std::cout << "Search time for " << size << " is: " << time << std::endl;
}

int binarySearch(int arr[], int value, int left, int right) {

    while (left <= right) {

        int middle = (left + right) / 2;

        if (arr[middle] == value)

            return middle;

        else if (arr[middle] > value)

            right = middle - 1;

        else

            left = middle + 1;

    }

    return -1;

}

File: Assignment1/Problem5/Java/src/Main.java
import java.util.Arrays;
public class Main {

    public static void main(String[] args) {
	    int size = 128;
        int[] array = new int[size];
        for (int i = 0; i<size; i++){
            array[i]=1;
        }
        long start = System.nanoTime();
        for (int i = 0; i < 10000000; i++){
            Arrays.binarySearch(array, 0);
        }
        long end = System.nanoTime();
        long run = end - start;
        System.out.println("Execution time for " + size + " is: " + run);
    }

    /*public static int search(int[] array, int target){
        int first = 0;
        int last = array.length -1;
        int mid;
        int position = -1;
        boolean found = false;

        while (!found && first <= last);{
            mid = (first+last)/2;
            if (array[mid]==target){
                found = true;
                position = mid;
            }
            else if (array[mid] > target){
                last = mid - 1;
            }
            else {
                first = mid + 1;
            }
        }

        return position;
    }*/
}
